cli/electron.ts

```diff
@@ -0,0 +1,280 @@
+let targetProductJson: string;
+let targetId: string;
+let isInit = false;
+let buildOut: string;
+
+function errorOut(msg: string): Promise<void> {
+    console.error(msg);
+    process.exit(1)
+
+    return null;
+}
+
+export function electronAsync(parsed: p.ParsedCommand): Promise<void> {
+    // Ensure there is a subcommand
+    let subcommand = parsed.arguments[0];
+
+    if (!subcommand) {
+        errorOut("Please specify a subcommand");
+    }
+
+    // Validate current target
+    let needsCurrentTarget = (subcommand !== "build" && subcommand !== "dist") || !parsed.flags["release"];
+
+    if (needsCurrentTarget && !fs.existsSync("pxtarget.json") && !fs.existsSync("package.json")) {
+        errorOut("This command requires to be in a valid target directory (pxtarget.json and package.json required)");
+    }
+
+    targetId = JSON.parse(fs.readFileSync("package.json", "utf8")).name;
+
+    // Find root of PXT Electron app sources
+    if (parsed.flags["pxtElectron"]) {
+        pxtElectronPath = parsed.flags["pxtElectron"] as string;
+
+        if (!fs.existsSync(pxtElectronPath)) {
+            errorOut("Cannot find the specified PXT Electron app: " + pxtElectronPath);
+        }
+    } else {
+        pxtElectronPath = path.join("..", "pxt", "electron");
+
+        if (!fs.existsSync(pxtElectronPath)) {
+            errorOut("Please specify --pxtElectron path; cannot find PXT Electron at default location: " + pxtElectronPath);
+        }
+    }
+
+    pxtElectronPath = path.resolve(pxtElectronPath);
+    pxtElectronSrcPath = path.resolve(pxtElectronPath, "src");
+
+    // Ensure there is a product.json
+    if (subcommand !== "init") {
+        if (parsed.flags["product"]) {
+            targetProductJson = parsed.flags["product"] as string;
+
+            if (!fs.existsSync(targetProductJson)) {
+                errorOut("Cannot find the specified product.json file: " + targetProductJson);
+            }
+        } else {
+            targetProductJson = path.join("electron", "product.json");
+
+            if (!fs.existsSync(targetProductJson)) {
+                errorOut("Please specify --product path; cannot find product.json at default location: " + targetProductJson);
+            }
+        }
+
+        targetProductJson = path.resolve(targetProductJson);
+    }
+
+    // Other initializations
+    let linkedTarget = path.join(pxtElectronSrcPath, "node_modules", targetId);
+    let linkPath = fs.existsSync(linkedTarget) ? finalLinkPath(linkedTarget) : null;
+
+    isInit = path.resolve(linkPath) === path.resolve(process.cwd());
+
+    if (parsed.flags["release"]) {
+        buildOut = "out";
+    } else {
+        buildOut = path.join(process.cwd(), "electron-out");
+    }
+
+    // Invoke subcommand
+    switch (subcommand) {
+        case "init":
+            return initAsync();
+        case "run":
+            return runAsync();
+        case "build":
+            return buildAsync(parsed);
+        case "dist":
```

Comment:
I use "package" and "build" interchangeably in comments, code and readme, so what about "buildInstaller" or something?